app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt

```diff
@@ -236,23 +244,50 @@ class AppVersionActivityTest {
   @Test
   fun testAppVersionActivity_loadFragment_onBackPressed_displaysAdministratorControlsActivity() {
     ActivityScenario.launch<AdministratorControlsActivity>(
-      launchAdministratorControlsActivityIntent(
-        internalProfileId = 0
-      )
-    ).use {
+      launchAdministratorControlsActivityIntent(internalProfileId = 0)
+    ).use { scenario ->
+
       testCoroutineDispatchers.runCurrent()
+
+      // Scroll to the item in the RecyclerView
       onView(withId(R.id.administrator_controls_list)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(
-          3
-        )
+        scrollToPosition<RecyclerView.ViewHolder>(3)
       )
+
+      // Click on the app version option
       onView(withText(R.string.administrator_controls_app_version)).perform(click())
-      intended(hasComponent(AppVersionActivity::class.java.name))
-      onView(isRoot()).perform(pressBack())
+
+      // Wait for UI update
+      testCoroutineDispatchers.runCurrent()
+
+      // Verify if the AppVersionActivity is running
+      var currentActivity: Activity? = null
+      scenario.onActivity { activity ->
+        currentActivity = getCurrentActivity()
+        assertThat(currentActivity).isInstanceOf(AppVersionActivity::class.java)
+      }
+
+      // Simulate pressing back via activity rather than Espresso
+      (currentActivity as? AppVersionActivity)?.onBackPressed()
+
+      // Wait for UI update
+      testCoroutineDispatchers.runCurrent()
+
+      // Verify that AdministratorControlsActivity is visible again
       onView(withId(R.id.administrator_controls_list)).check(matches(isDisplayed()))
     }
   }
 
+  fun getCurrentActivity(): Activity? {
+    val activity = arrayOfNulls<Activity>(1)
+    InstrumentationRegistry.getInstrumentation().runOnMainSync {
+      activity[0] = ActivityLifecycleMonitorRegistry.getInstance()
+        .getActivitiesInStage(Stage.RESUMED)
+        .firstOrNull()
+    }
+    return activity[0]
+  }
```

Comment:
We currently don't use this implementation in any of our UI tests, and there is probaby an existing pattern that you can follow to achieve the same objective as here. Please search for any tests that use `ActivityScenario.launch` to see the implementation patterns used.

Reply:
This function is unused, please delete.